parse_test.go

```diff
@@ -45,6 +45,32 @@ func TestParse(t *testing.T) {
 	}
 }
 
+func TestParseUntilEOF(t *testing.T) {
+	files, err := ioutil.ReadDir("./testdata")
+	if err != nil {
+		t.Fatalf("unable to read testdata/: %v", err)
+	}
+	for _, f := range files {
+		if !f.IsDir() && strings.HasSuffix(f.Name(), ".dcm") {
+			t.Run(f.Name(), func(t *testing.T) {
+				dcm, err := os.Open("./testdata/" + f.Name())
+				if err != nil {
+					t.Errorf("Unable to open %s. Error: %v", f.Name(), err)
+				}
+				defer dcm.Close()
+
+				if err != nil {
+					t.Errorf("Unable to stat %s. Error: %v", f.Name(), err)
+				}
+				_, err = dicom.ParseUntilEOF(dcm, nil)
```

Comment:
Since ParseUntilEOF is not covered by the roundtrip tests [here](https://github.com/suyashkumar/dicom/blob/f77ffdec9a83ffddf27d05aeaeb97f7fb220e589/write_test.go#L30), I think we should update this test to compare the dataset between ParseUntilEOF and Parse implementations for each test file and ensure they are identical. This would add some degree of conformance checking that would be valuable.

WDYT?

Reply:
Can you address this as well? Or, if you'd like, I can do so by adding a commit to your branch or sending a PR to your branch to address this.

Reply:
I did a quick pass at this and opened https://github.com/amitbet/dicom/pull/1/files. If you merge that into your contrib branch, I think the commit should automatically show up on this PR as well.